deepchecks/recommender/checks/product_association.py

```diff
@@ -0,0 +1,165 @@
+        association_counts['lift_metric (%)'] = association_counts['co-occurence']/(association_counts[f'prev_{item_col}_prb']*association_counts[f'{item_col}_prb']*len(filtered_interactions)).astype(np.float16)
+        
+        association_counts = association_counts[[f'prev_{item_col}', item_col,'co-occurence','lift_metric (%)']]
+        movie_translation = item_df[[item_col,item_column_name]].set_index(item_col)[item_column_name].to_dict()
+        
+        association_counts[f'prev_{item_col}']= association_counts[f'prev_{item_col}'].map(movie_translation)
+        association_counts[item_col]= association_counts[item_col].map(movie_translation)
+
+        association_counts = association_counts.sort_values(['co-occurence','lift_metric (%)'],ascending=[False,False]).rename({f'prev_{item_col}':'reference',
+        item_col:'possible recommandation'},axis=1)
+        display_df = association_counts.head(20)
+        # Create a network graph
+        G = nx.Graph()
+
+        # Add edges to the graph based on the lift metric above the threshold
+        for _, row in display_df.iterrows():
+                G.add_edge(row['reference'], row['possible recommandation'], weight=row['lift_metric (%)'])
+
+        # Set node positions using a spring layout
+        pos = nx.spring_layout(G,k=1.2,seed=32)
+
+
+        # Create Plotly nodes
+        nodes_trace = go.Scatter(
+            x=[pos[node][0] for node in G.nodes()],
+            y=[pos[node][1] for node in G.nodes()],
+            mode='markers+text',
+            hovertext=list(G.nodes()),
+            textposition='top center',
+            hoverinfo='text',
+            marker=dict(
+                showscale=False,
+                color=list(dict(G.degree).values()),
+                size=20,
+                line_width=2
+            )
+        )
+
+        # Create Plotly edges
+        edge_x = []
+        edge_y = []
+        edge_weights = []
+        for edge in G.edges():
+            x0, y0 = pos[edge[0]]
+            x1, y1 = pos[edge[1]]
+            edge_x.extend([x0, x1, None])
+            edge_y.extend([y0, y1, None])
+            edge_weights.append(G.edges[edge]['weight'])
+
+        edges_trace = go.Scatter(
+            x=edge_x,
+            y=edge_y,
+            line=dict(width=0.5, color='gray'),
+            textposition='top center',
+            hoverinfo='text',  # Show hovertext
+            mode='lines'
+        )
+
+        # Combine nodes, edges, and edge labels traces
+        fig = go.Figure(data=[edges_trace, nodes_trace])
+
+        # Set plot layout
+        fig.update_layout(
+            title_text='Item Associations based on Lift Metric',
+            showlegend=False,
+            hovermode='closest',
+            margin=dict(b=0, l=0, r=0, t=40),
+        )
+
+        # Show the interactive plot
+
+        return CheckResult(association_counts, header='Product Association', display=[fig])
```

Comment:
I think that it's legit here to add a short text explaining what we're seeing in this graph

Reply:
fixed:
https://github.com/deepchecks/deepchecks/commit/fa0e410dce9cd742db9fb857414ac3af4676873e